notebooks/analysis/nlm_text_based_analysis.ipynb

```diff
@@ -254,6 +254,94 @@
     "    'doc_vector_lengths')"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Below is what the equivalent inputs are in our case\n",
+    "# term_frequencies, doc_frequencies, abstract_bag_of_words, query, doc_vector_lengths, false\n",
+    "def cosineScore(tfPostingList, dfPostingList, terms, query, docVectorLengths, alternativeWeighting=False):\n",
+    "    scores = dict(zip([_['nlm_unique_id'] for _ in terms], [0 for _ in terms]))\n",
+    "    totalCorpusDocCount = len(scores)\n",
+    "\n",
+    "    modifiedQuery = [{'docID': 0, 'doc': query}]\n",
+    "    modifiedQueryTerms = get_bag_of_words_from_corpus(modifiedQuery)\n",
+    "    tf_tqs = get_term_frequencies(modifiedQuery, modifiedQueryTerms)\n",
```

Comment:
What is this actually doing? Do we need the original corpus to run this? Look into how cosineScores are setup?